lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -59,7 +59,7 @@ def capture(money, authorization, options={})
       def refund(money, authorization, options={})
         post = init_post(options)
         add_invoice_for_modification(post, money, authorization, options)
-        add_references(post, authorization, options)
+        add_references(post, psp_reference_from(authorization), options)
```

Comment:
This is a mistake.
I'll have to revisit. It should just be `authorization` and not `psp_reference_from(authorization)`, since `add_references` already does that.